### PR TITLE
add possibility to specify multiple keys at once (-k key1 -k key2 ...)

### PR DESCRIPTION
**Pull request by jonny08152**
New Feature :boom:

### CODE REVIEW

**Comment by sebidude** [comment]
I like the idea, but it introduces a confusing behavior when converting from a yaml file with encrypted values to a kubernetes secret. If mutliple keys are specified to find the maps in the yaml file and a key with same name exists in more than one map, the last one will win and the kubernetes secret will not state all the data. 

I suggest to not convert to a kubernetes secret if multiple keys are specified and within the maps behind those keys, keys are same.

Here is how I tested:
```
cat << EOF > values.yaml 
data:
  password: testme
  token: foo
data2:
  password: test2
  token: foo2
safe:
  username: foobary
EOF

./kubecrypt yaml -e -i values.yaml -k data -k data2 -o safe.yaml
./kubecrypt convert foobar --key data -i safe.yaml 
./kubecrypt convert foobar --key data -k data2 -i safe.yaml 

```

It looks fine for a yaml file like this:
```
data:
  password: testme
  token: foo
data2:
  password2: test2
  token2: foo2
safe:
  username: foobary
```

**Comment by jonny08152** [comment]
Hmm I see. A solution for this would be to limit the "multiple keys" functionality to the **get** and **yaml** sub-commands even if that introduces some inconsistency in sub-command behaviors. But I think the benefits of having the functionality in those commands outweighs the disadvantages of introduced inconsistencies.

**Comment by sebidude** [comment]
What about checking for duplos and stop if a key is already stated in the secrets data?

**Comment by sebidude** [comment]
Ok. Let's not allow multiple keys for convert.
I'll do a review and you can do the fix then.

File: cmd/kubecrypt/main.go [hunk 1]
```diff
@@ -75,13 +75,13 @@ func main() {
...
 
-	yml.Flag("key", "Key in the yaml to be processed").Required().Short('k').StringVar(&keyname)
+	yml.Flag("key", "Keys in the yaml to be processed").Required().Short('k').StringsVar(&keynames)
 	yml.Flag("ecrypt", "Encrypt the values").Short('e').BoolVar(&encrypt)
 
 	convert := app.Command("convert", "convert encrypted yaml data to secret. If -e is passed create a yaml map with encrypted values of the data of the kubernetes secret.")
 	convert.Arg("secretname", "Name for the converted secret.").Required().StringVar(&secretname)
 	convert.Flag("encrypt", "Encrypt the values (default decrypt").Short('e').BoolVar(&encrypt)
-	convert.Flag("key", "Key in the yaml to be used as data for the secret").Required().Short('k').StringVar(&keyname)
+	convert.Flag("key", "Keys in the yaml to be used as data for the secret").Required().Short('k').StringsVar(&keynames)
```
Comment (sebidude) [inline cmd/kubecrypt/main.go]: Do not allow multiple keys here as discussed.

**Review by sebidude — CHANGES_REQUESTED** [review]

**Comment by jonny08152** [comment]
multiple keys are now only allowed for **yaml** and **get** sub commands

File: cmd/kubecrypt/main.go [hunk 2]
```diff
@@ -163,18 +164,15 @@ func main() {
 		yamldata, _ := processYamlData(encrypt, inputbytes)
 		writeOutputToFile(yamldata)
 	case "convert":
+		keynames = append(keynames, keyname)
```
Comment (sebidude) [inline cmd/kubecrypt/main.go:167]: Why do we need the keynames here?

**Review by sebidude — CHANGES_REQUESTED** [review]

File: cmd/kubecrypt/main.go (hunk 2 above)
Comment (jonny08152) [reply cmd/kubecrypt/main.go:167]: We could do it without that line but had to differentiate for one key vs. several keys in **processYamlData()** then.

Comment (sebidude) [reply cmd/kubecrypt/main.go:167]: got it.

**Pull request merged**